doc/Container Hardening/Reducing_Boot_Time_in_SONiC_by_Replacing_Process_Manager_HLD.md

```diff
@@ -0,0 +1,329 @@
+swss:fdbsyncd              OK        Process
+...
+```
+
+To achieve this functionality, the system-health checker has been enhanced to handle both supervisorctl and runit. It uses separate classes depending on the system configuration, validated by checking the Docker container label "sonic_docker_config"
+
+service status can be fetched using following commands
+
+```
+sv status /etc/service/*
+run: /etc/service/buffermgrd: (pid 84) 2394s
+run: /etc/service/orchagent: (pid 99) 2394s
+```
+
+
+### 4.9 Disabling runit and Re-enabling Supervisord
+Removing /etc/runit-manager and then rebooting (or executing config reload) will restore Supervisord as the process manager. The system will then launch containers configured to use supervisord, effectively reversing the migration to runit.
+
+
+```
+rm /etc/runit-manager
+reboot [or] config reload
+```
+
+## 5. SAI API
+N/A - This change is not directly related to the SAI API.
+
+## 6. Configuration and management
+### 6.1. Manifest (if the feature is an Application Extension)
+N/A -  Not applicable if not implemented as an application extension.
+
+### 6.2. CLI Enhancements
+```
+config process-manager runit <enabled/disabled>
+
+show process-manager runit 
+```
+
+### 6.3. Config DB Enhancements
+New table PROCESS-MANAGER will be introduced to hold if runit is enabled or disabled.
+```
+{
+    "PROCESS_MANAGER": {
+        "runit": {
+            "enabled": "true" 
+        }
+      }
+}
+```
+
+## 7. Warmboot and Fastboot Design Impact
+N/A
+
+## 8. Restrictions/Limitations
+Any limitations or restrictions imposed by the chosen process manager will be documented.
+
+## 9. Testing Requirements/Design
+### 9.1 Unit Test cases
+1. Verify whether all process starts up in all dockers
+2. Verify when critical process exits docker are restarted
+3. Verify system-healthd command works fine with runit
+4. Verify moving from supervisord to runit works fine
+5. Verify moving from runit to supervisord works fine
+6. Verify boot time improved with runit
+7. config reload and system reboot works fine
+8. Verify traffic loss improvement with runit after device reload.
+9. Verify warmboot and fastboot works fine with runit.
+
+### 9.2 System Test cases
+Comprehensive system tests will be conducted to measure boot time improvements and ensure the stability and functionality of SONiC with the new process manager.
+
+## 10. Open/Action items - if any
+Any open issues or action items will be tracked here.  This may include tasks like benchmarking different process managers, developing the configuration conversion tool, and updating the init process.
+
+```
+Currently, runit doesn't offer equivalent functionality to Supervisord supervisor-proc-exit-listener for syslog alerting based on process states. This is a gap in functionality we need to address.
+
+Restart of docker derived based on auto-restart attribute in Feature table is not currently handled and will be handled later.
```

Comment:
Container lifetime is not controlled by init process inside container, coudl you please clarify?